lib/r10k/source/base.rb

```diff
@@ -65,4 +65,17 @@ def accept(visitor)
       end
     end
   end
+
+  private
+
+  def derive_prefix(name,prefix)
```

Comment:
I thought I commented on this before but it doesn't look like I did; since this deals with branch name munging I would like this responsibility to be moved to the BranchName class so all the name manipulation is in a single place.

Given this change, the type of `@prefix` will be `[Boolean, String]`.